SPECS/mozjs/mozjs.spec

```diff
@@ -1,159 +1,307 @@
+
+# LTO - Enable in Release builds, but consider disabling for development as it increases compile time
+%global build_with_lto    1
+
+# Require tests to pass?
+%global require_tests     1
+
+%if 0%{?build_with_lto}
+# LTO is the default
+%else
+%define _lto_cflags %{nil}
+%endif
+
+# Disable LTO on aarch64 (borked since GCC14)
+# https://bugzilla.redhat.com/show_bug.cgi?id=2260867
+%ifarch aarch64
+%define _lto_cflags %{nil}
+%endif
+
+# Big endian platforms
+%ifarch ppc ppc64 s390 s390x
+%global big_endian 1
+%endif
 
-Summary:        Mozilla's JavaScript engine.
 Name:           mozjs
-Version:        78.10.0
-Release:        5%{?dist}
-License:        BSD AND MIT AND MPLv2.0 AND Unicode
+Version:        102.15.1
+Release:        1%{?dist}
+Summary:        SpiderMonkey JavaScript library
+
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
-Group:          Applications/System
+
+License:        MPL-2.0 AND Apache-2.0 AND BSD-3-Clause AND BSD-2-Clause AND MIT AND GPL-3.0-or-later
 URL:            https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey
 Source0:        https://ftp.mozilla.org/pub/firefox/releases/%{version}esr/source/firefox-%{version}esr.source.tar.xz
-Patch0:         emitter.patch
-Patch1:         emitter_test.patch
+
+# Known failures with system libicu
+Source1:        known_failures.txt
+
+# Patches from mozjs68, rebased for mozjs78:
+Patch01:        fix-soname.patch
+Patch02:        copy-headers.patch
+Patch03:        tests-increase-timeout.patch
+Patch09:        icu_sources_data.py-Decouple-from-Mozilla-build-system.patch
+Patch10:        icu_sources_data-Write-command-output-to-our-stderr.patch
+
+# Build fixes - https://hg.mozilla.org/mozilla-central/rev/ca36a6c4f8a4a0ddaa033fdbe20836d87bbfb873
+Patch12:        emitter.patch
+Patch13:        tests-Use-native-TemporaryDirectory.patch
+
 # Build fixes
-Patch2:         init_patch.patch
-Patch3:         spidermonkey_checks_disable.patch
-Patch4:         fix-soname.patch
-Patch5:         CVE-2022-48285.patch
+Patch14:        init_patch.patch
+Patch15:        remove-sloppy-m4-detection-from-bundled-autoconf.patch
+
+# tentative patch for RUSTFLAGS parsing issue, taken from firefox package:
+# https://bugzilla.redhat.com/show_bug.cgi?id=2184743
+# https://bugzilla.mozilla.org/show_bug.cgi?id=1474486
+Patch16:        firefox-112.0-commasplit.patch
+
+Patch17:        Replace-distutils.LooseVersion-with-looseversion.LooseVersion.patch
+# This is ugly, I know, I know...
+Patch18:        Hacky-workaround-for-expected-string-or-bytes-like-object.patch
 
-BuildRequires:  autoconf213
+# TODO: Check with mozilla for cause of these fails and re-enable spidermonkey compile time checks if needed
+Patch20:        spidermonkey_checks_disable.patch
+
+# s390x/ppc64 fixes
+Patch21:        0001-Skip-failing-tests-on-ppc64-and-s390x.patch
+
+# AzLinux CVE patches
+Patch30:       CVE-2022-48285.patch
+
+%if !0%{?azl}
+BuildRequires:  cargo
+%endif
```

Comment:
Why even keep these BRs, if we don't use them? Also, why `cargo` is no needed for Azure Linux?

Reply:
Makes sense. An extra comment may help, no strong push.

Reply:
Added comment